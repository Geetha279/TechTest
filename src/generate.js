import React from 'react';
import './App.css';
import { connect } from 'react-redux';

export class Generate extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            pin: [],
            msg: '',

        }
    };
    handleChange() {

        var newArray = [];
        for (var i = 0; i < 5; ++i) {
            var val = Math.floor(1000 + Math.random() * 9001);

            newArray.push(val);
        }
        this.setState({ pin: newArray });


        this.setState({
            msg: 'Name',
        });

    };
    render() {

        console.log("zzz", this.state.pin);
        return (

            <React.Fragment>
                <div className="Generate">
                    <div>
                        <table className="TableStyle"><tbody><tr>{this.state.pin.length > 0 ? this.state.pin.map(item => <td className="tablecellstyle" key={item}>{item}</td>) : <td></td>}</tr></tbody></table>
                    </div>
                    <div style={{ marginTop: '40px' }}>
                        <button type="submit" className="button" onClick={this.handleChange.bind(this)}> GENERATE </button>
                        <button type="submit" onClick={() => this.props.onStoreCredentials(this.state)} className="savebutton"> SAVE </button>
                    </div>
                </div>

            </React.Fragment>
        );
    }
}

const mapDispatchToProps = dispatch => {
    console.log("inside mapdispatch ");
    return {
        onStoreCredentials: (response) => dispatch({

            type: "pin",
            result: response
        })

    }
}

export default connect(null, mapDispatchToProps)(Generate);




