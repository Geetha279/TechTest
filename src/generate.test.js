import React from 'react';
import { shallow, configure } from 'enzyme';
import { Generate } from './generate';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Generate component', () => {
    it('check div', () => {
        const wrapper = shallow(<Generate />);
        expect(wrapper).toMatchSnapshot();
    });
});




