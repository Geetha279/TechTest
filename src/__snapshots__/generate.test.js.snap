// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`Generate component check div 1`] = `ShallowWrapper {}`;
